backend/core/management/commands/delete_inactive_users.py

```diff
@@ -0,0 +1,46 @@
+"""
+Delete users that have not activated their account.
+"""
+import datetime
+
+from django.core.management.base import BaseCommand
+from django.utils import timezone
+
+from core.models.user import User
+from settings import settings
+
+
+class Command(BaseCommand):
+    help = "Delete users that have not activated their account."
+
+    def handle(self, *args, **options):
+        self.stdout.write(f"start command: {__name__}")
+
+        days_to_keep = settings.APP_CORE["MGMT_DELETE_INACTIVE_USERS_PERIOD"]
+        delete_before = timezone.now() - datetime.timedelta(days=days_to_keep)
+
+        n_users_to_delete = User.objects.filter(
+            is_active=False, date_joined__lt=delete_before
+        ).count()
+
+        # display the configuration if more verbosity is asked
+        if options.get("verbosity", 1) > 1:
+            self.stdout.write(f"MGMT_DELETE_INACTIVE_USERS_PERIOD: {days_to_keep}")
+
+        self.stdout.write(f"{n_users_to_delete} old inactive users found")
+        deleted = User.objects.filter(
+            is_active=False, date_joined__lt=delete_before
+        ).delete()
```

Comment:
Any reason to count and delete use users separately? Technically, the count may change between the 2 queries. And I think any problem during the delete would cause an exception, so the condition `n_users_to_delete != deleted[0]` can only be True when the count has changed in the meantime, right?

Reply:
My original idea was to show in the logs that among this number of users, only this number have been deleted.

I think this is not that relevant if indeed the `.delete()` method fails completely in case of error on an item, as an alert will be raised on Discord. I don't know exactly how `.delete()` works here, the documentation says that its implementation may be subject to change.

In any case, if `.delete()` fails completely in case of error, it could be useful to display in the logs number of users to delete. Maybe I can read the code of `.delete()` and see if there are any obvious exceptions that are raised.

Reply:
It's not clear what exceptions could be raised by `.delete()`.

According to the Django source code, it could be `TypeError` if the query set is misconfigured.
[See here](https://github.com/django/django/blob/06ebaa9e287137d4496a2736b055f1c6aed95839/django/db/models/query.py#L937)

Or it could be any exceptions raised by the `SQLCompiler.connection.cursor.execute` method.
[See here](https://github.com/django/django/blob/06ebaa9e287137d4496a2736b055f1c6aed95839/django/db/models/sql/compiler.py#L1381)

To not add unnecessary complexity, I think we could wait for an exception to be raised to see exactly what we should expect. In fact, it's even not necessary to catch an exception here, but I wanted to log the remaining number of users to delete still in the database, which require to catch the exception.

Reply:
Fixed by https://github.com/tournesol-app/tournesol/pull/799/commits/a7cf57fe84063fe0469ce5f6c231758872c9e26f and https://github.com/tournesol-app/tournesol/pull/799/commits/c6bc698c0e03aba0e0aeba1e1501fe9456d3a96b.